pkg/api/types.go

```diff
@@ -1004,6 +1008,30 @@ type Service struct {
 	Status ServiceStatus `json:"status,omitempty"`
 }
 
+// ServiceAccount binds together:
+// * a name, understood by users, and perhaps by peripheral systems, for an identity
+// * a principal that can be authenticated and authorized
+// * a security context, which defines the Linux Capabilities, User IDs, Groups IDs, and other capabilities and controls on interaction with the file system and OS.
+// * a set of secrets
+type ServiceAccount struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// APISecret is the name of the secret to use to identify this ServiceAccount to the API
+	// It must be of type SecretTypeServiceAccountToken
+	APISecret string `json:"apiSecret"`
+	// Secrets is the list of secrets allowed to be used by pods running using this ServiceAccount
+	Secrets []string `json:"secrets"`
```

Comment:
Would some clusters ever have (I'm making this but, hopefully you get the idea) a GithubSecretCrontroller which adds Github keys to pods based on a certain default github project for the namespace.  Or a GoogleCloudStorageController which adds GoogleCloudStorage bucket keys.  Etc.

If that is the case, then those initializers will need to be allowed to edit the Secrets array.  Which requires that the array have a `patchStrategy:"merge" patchMergeKey:"foo"`.

What would foo be?  

If we aren't sure what foo is, then I think this array needs to be a map.